tests/fixtures/controllers/methodController.ts

```diff
@@ -71,6 +71,13 @@ export class MethodController extends Controller {
         return Promise.resolve();
     }
 
+    @SuccessResponse<TestModel>('201', 'Created')
+    @Get('typedSuccessResponse')
+    public async typedSuccessResponse(): Promise<void> {
+        this.setStatus(201);
+        return Promise.resolve();
```

Comment:
This example clearly states why I think this is a dangerous approach. Your swagger model is going to say that you API returns an object with the shape `TestModel` but your code does not actually return a `TestModel` at all. To me this defeats the purpose of using TypeScript. I apologize, but I really can't accept this PR. However, now that I understand your use case better, I believe what you're looking for is the ability to clearly define the the shape of the error.

> As a user I'd want to know why my requests fail. As a developer, when I know I can't handle a request, I want to not just throw an error and exit but have a well defined and documented response through swagger (with example).
> Writing typescript, I'd want to retrieveModel(...conditions): Model | ModelNotFound.
> In swagger, both these things should be documented, the 200 with the model and the 404 ModelNotFound with specific reasons why the model was not found.

To accomplish `Model | ModelNotFound` the best way to accomplish that translation between the union type and the swagger model is to implement `$OneOf`. So I'm going to close this PR, and then if you or someone wants to pick up issue https://github.com/lukeautry/tsoa/issues/393, then it would be much easier to accomplish the use case you're asking for. But I can't approve a PR that would cause the swagger model to be out of sync with the code since that's the mission statement of tsoa.